extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemLockupExtractor.java

```diff
@@ -0,0 +1,278 @@
+        return metadataPart(1, 1)
+            .map(this::getTextContentFromMetadataPart)
+            .orElse(null);
+    }
+
+    @Nullable
+    @Override
+    public DateWrapper getUploadDate() throws ParsingException {
+        if (timeAgoParser == null) {
+            return null;
+        }
+
+        return timeAgoParser.parse(getTextualUploadDate());
+    }
+
+    @Override
+    public long getViewCount() throws ParsingException {
+        if (isPremium() || isPremiere()) {
+            return -1;
+        }
+
+        // TODO Check if this is the same for shorts
+        final Optional<String> optTextContent = metadataPart(1, 0)
+            .map(this::getTextContentFromMetadataPart);
+        // We could do this inline if the ParsingException would be a RuntimeException -.-
+        if (optTextContent.isPresent()) {
+            return getViewCountFromViewCountText(optTextContent.get());
+        }
+        return -1;
+    }
+
+    protected long getViewCountFromViewCountText(@Nonnull final String viewCountText)
+            throws NumberFormatException, ParsingException {
+        // These approaches are language dependent
+        if (viewCountText.toLowerCase().contains(NO_VIEWS_LOWERCASE)) {
+            return 0;
+        } else if (viewCountText.toLowerCase().contains("recommended")) {
+            return -1;
+        }
+
+        return Utils.mixedNumberWordToLong(viewCountText);
+    }
+
+    @Nonnull
+    @Override
+    public List<Image> getThumbnails() throws ParsingException {
+        return YoutubeParsingHelper.getImagesFromThumbnailsArray(
+            JsonUtils.getArray(lockupViewModel,
+                "contentImage.thumbnailViewModel.image.sources"));
+    }
+
+    protected boolean isPremium() {
+        // TODO Detect with samples
+        return false;
+    }
+
+    protected boolean isPremiere() {
+        // TODO Detect with samples
+        return false;
+    }
+
+    protected Optional<JsonObject> metadataPart(final int rowIndex, final int partIndex)
+        throws ParsingException {
+        return JsonUtils.getArray(lockupViewModel,
+                "metadata.lockupMetadataViewModel.metadata"
+                    + ".contentMetadataViewModel.metadataRows")
+            .streamAsJsonObjects()
+            .skip(rowIndex)
+            .limit(1)
+            .flatMap(jsonObject -> jsonObject.getArray("metadataParts")
+                .streamAsJsonObjects()
+                .skip(partIndex)
+                .limit(1))
+            .findFirst();
+    }
+
+    protected String getTextContentFromMetadataPart(final JsonObject metadataPart) {
+        return metadataPart.getObject("text").getString("content");
+    }
+
+    @Nullable
+    @Override
+    public String getShortDescription() throws ParsingException {
+        return null;
+    }
+
+    @Override
+    public boolean isShortFormContent() throws ParsingException {
+        // TODO Detect with samples
```

Comment:
Shorts seem to be still returned as `compactVideoRenderer`s, I got two of them within `lockupViewModels`.

Reply:
I tried to find a video with shorts recommendations but was unable to find a single one (just compilations of shorts lol).
Could you share some example data? :)